src/prep.py

```diff
@@ -0,0 +1,331 @@
+            "Example Usage: $ cvise test.sh string.i"
+        )
+    except:
+        warn(
+            f"Couldn't change permissions of {script_path} to enable execution. "
+            "Do this manually."
+        )
+
+
+def double_check_removal_with_user(*, file: Path, force_rm: bool) -> None:
+    # TODO: add optional rename
+    while not force_rm:
+        warning_msg = warn(
+            f"target {file} already exists. Remove it? (y/n): ", ret_only=True
+        )
+        user_inp = input(warning_msg)
+        user_inp = user_inp.strip().lower()
+        if user_inp == "y":
+            break
+        elif user_inp == "n":
+            success("OK: quitting.")
+            sys.exit(0)
+    info(f"Removing {file} which already exists")
+    file.unlink()
+
+
+def setup_argparser(parser: argparse.ArgumentParser) -> None:
+    """
+    Parse out arguments from command line interface.
+    Use various validate_foo_from_cli functions to assist end-user
+    """
+    global DEBUG
+
+    parser.add_argument(
+        "build_command",
+        help="The specific build command preamble used sans the target. "
+        "Example: `make -j$(nproc) LLVM=1 V=1`",
+        nargs="*",
+        default=["make", f"-j{os.cpu_count()}", "LLVM=1", "V=1"],
+    )
+
+    parser.add_argument(
+        "target",
+        help="The source file you wish to target. Example: lib/string.o",
+        type=validate_target_from_cli,
+    )
+
+    parser.add_argument(
+        "-p",
+        "--path-to-linux",
+        type=lambda arg: validate_path_from_cli(arg, "-p/--path-to-linux"),
+        default="./",
+        help="Specify the Linux source tree directory",
+    )
+
+    parser.add_argument(
+        "-f",
+        "--force-rm-existing-target",
+        help="If the target (foo.o) already exists, remove it -- without prompting",
+        action="store_true",
+        default=False,
+    )
+
+    parser.add_argument(
+        "-F",
+        "--force-rm-existing-script",
+        help="If test.sh already exists, remove it -- without prompting",
+        action="store_true",
+        default=False,
+    )
+
+    parser.add_argument(
+        "-d",
+        "--debug",
+        help="Enable prepreduce Debug logs",
+        action="store_true",
+        default=False,
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="Output directory of prepreduce outputs (.i files and test scripts)",
+        default=os.getcwd(),
+        type=lambda arg: validate_path_from_cli(arg, "-o/--output", allow_mkdir=True),
+    )
+
+    parser.add_argument(
+        "--no-go-fast",
+        help="Disable -Wfatal-errors (will slow down cvise greatly)",
+        action="store_true",
+        default=False,
+    )
+
+
+def validate_cli_args(cli_args):
+    DEBUG = cli_args.debug
+    if DEBUG:
+        ic.enable()
+
+    validate_build_cmd_from_cli(cli_args.build_command)
+
+
+def is_kernel_configured_for_clang(path_to_linux: Path) -> Optional[bool]:
```

Comment:
an optional bool? is that really necessary?

Reply:
Probably not. Fixed.